Wikipedia/Code/SavedArticlesFetcher.m

```diff
@@ -206,10 +213,15 @@ - (AnyPromise*)fetchImageInfoForImagesInArticle:(MWKArticle*)article {
 
     if (imageFileTitles.count == 0) {
         DDLogVerbose(@"No image info to fetch, returning successful promise with empty array.");
-        return [AnyPromise promiseWithValue:imageFileTitles];
+        success(imageFileTitles);
+        return;
     }
 
-    return PMKJoin([[imageFileTitles bk_map:^AnyPromise*(NSString* canonicalFilename) {
+    for (NSString *canonicalFilename in imageFileTitles) {
+        [self.imageInfoFetcher fetchGalleryInfoForImage:canonicalFilename fromSite:article.title.site];
```

Comment:
Opps this is fetching *gallery info*... sorry if that last comment isn't relevant :)